src/pages/Integrations/SplunkSetup/Constants.ts

```diff
@@ -0,0 +1,23 @@
+export const RBAC_GROUPS_API = '/api/rbac/v1/groups/';
+export const RBAC_ROLES_API = '/api/rbac/v1/groups/{}/roles/';
+export const RBAC_ADD_USER_API = '/api/rbac/v1/groups/{}/principals/';
+export const CREATE_INTEGRATION_API = '/api/integrations/v1.0/endpoints';
+export const CREATE_BEHAVIOR_GROUP_API = '/api/notifications/v1.0/notifications/behaviorGroups';
+export const UPDATE_BEHAVIOR_GROUP_API = '/api/notifications/v1.0/notifications/behaviorGroups/{}/actions';
+export const EVENTTYPE_API = '/api/notifications/v1.0/notifications/eventTypes/{}/behaviorGroups';
```

Comment:
We alredy have these in a typesafe API
Could you take a look at the [fetching library](https://marcin-piela.github.io/react-fetching-library
) we use? 

- [Create Integration](https://github.com/RedHatInsights/notifications-frontend/blob/master/src/services/useSaveIntegration.ts#L37)
- [Create/Update behavior group](https://github.com/RedHatInsights/notifications-frontend/blob/master/src/services/Notifications/SaveBehaviorGroup.ts)
- [Events](https://github.com/RedHatInsights/notifications-frontend/blob/master/src/services/EventLog/GetNotificationEvents.ts)

Let me know if you have questions

Reply:
The way it is written is more functional. The library heavily uses hooks to make queries which need to be bound to a component on a top level. I'm looking at a way to migrate to it, but if you @josejulio would have an idea I'd love to hear it.